etc/examples/forwarders/http-api-tts-ttn.ini

```diff
@@ -0,0 +1,82 @@
+; ------------------------------------------
+; Family:   Protocol forwarder
+; About:    Versatile protocol forwarder components for bridging the gap between
+;           different data sinks, bus systems and serialization formats.
+; ------------------------------------------
+; Name:     TTS/TTN-to-MQTT protocol forwarder
+; About:    Suitable for data acquisition via TTS/TTN Webhook HTTP POST requests.
+; Channel:  Transport: HTTP, MQTT; Format: JSON
+; See also: https://getkotori.org/docs/handbook/forwarders/tts-ttn.html
+;           https://getkotori.org/docs/handbook/acquisition/protocol/http.html
+;           https://getkotori.org/docs/handbook/decoders/
+; ------------------------------------------
+; Description:
+;
+;         - Listen to HTTP POST or PUT requests
+;         - Receive payloads formatted as JSON or in urlencoded format
+;         - Decode payloads using specific decoders
+;         - Forward payloads to the MQTT bus
+;
+; Manual:   Please specify forwarding source and target parameters in URI format.
+;           The MQTT topic path is derived from the HTTP URI path by interpolating
+;           the appropriate part of the context URI.
+;
+; Example:  In the example below, given the "address" part of the resource URI
+;           is "testdrive/area-42/node-1", data sent to the full URIs
+;
+;               /api/mqttkit-1/ttn/{devID}
+;               /api/mqttkit-1/ttn/{devID}/uplinks
+;               /api/mqttkit-1/ttn/{devID}/join-accept
+;               /api/mqttkit-1/ttn/{devID}/downlink-ack
+;
+;           with, e.g.::
+;
+;               devID=mqttkit-1/testdrive/area-42/node-1
+;
+;           will be republished to the MQTT topics
+;
+;               mqttkit-1/testdrive/area-42/node-1/data.json
+;               mqttkit-1/testdrive/area-42/node-1/events.json
+;
+; ------------------------------------------
+
+
+[mqttkit-1.http-api-tts-ttn.composite]
+enable      = false
+type        = application
+realm       = mqttkit-1
+mqtt_topics = mqttkit-1/#
+app_factory = kotori.daq.application.composite:boot
+services    = kotori.daq.services.mig:MqttInfluxGrafanaService
+graphing    = kotori.daq.graphing.grafana:GrafanaManager
+strategy    = kotori.daq.strategy.tts_ttn:TheThingsWanBusStrategy
+
+
+
+[mqttkit-1.http-api-tts-ttn]
+enable          = true
+
+type            = application
+application     = kotori.io.protocol.forwarder:boot
+#app_factory     = kotori.daq.application.composite:boot
+
+realm           = mqttkit-1
+source          = http:/api/{realm:mqttkit-1}/ttn/{device_id:.*}/{slot:(uplinks|join-accept|downlink-ack)} [POST]
+target          = mqtt:/{realm}/{address}/ttn/{slot}.json
```

Comment:
Regarding our discussion at [TTS-/TTN-Daten an Kotori weiterleiten](https://community.hiveeyes.org/t/tts-ttn-daten-an-kotori-weiterleiten/1422/51) ff., whether to use a prefix like `/apittn`, or a suffix like `/data.ttn.json`, for signalling that it's a request from TTN, it looks like I've proposed a third variant here, using an infix like `/ttn` after the `<realm>` component. Interesting!

/cc @thiasB

Reply:
The reason I don't fancy the prefix variant too much is that the addressing slot at the very front of the URL is usually managed by infrastructure outside of Kotori. For example, we are using Nginx to dispatch exactly two prefix routes to Kotori, that is `/api`, and `/api-notls`. In order to keep that interface concise and lean, I think we should not use this particular addressing slot for other purposes than the two main entrypoints it is currently used for.

Thinking more about the topic in general, I am leaning towards the idea that signalling special decoding needs should actually be **pushed to the end of the URL instead**, so maybe we will finally introduce query parameters for that purpose, instead of trying to squeeze those metadata into the URL path itself?

In this case, we would not need any special forwarding component with an accompanying pattern-based forwarding rule configuration like outlined above, but would just signal additional information to the (passive) decoder component instead, which would be able to take two routes of dispatching, based on this information:

1. Apply decoding based on the "`device_id` in URL" variant.
2. Apply decoding based on the "`device_id` in Payload" variant.

On the other hand, it may be better to let the operator decide, and configure TTN capabilities on an existing channel group by adding a special flag to the configuration snippet. I am not yet 100% sure about that detail, this is why I am bouncing on an off about it. Personally, in order to give more power to the users, I am leaning towards the hassle-free "works out of the box" variant, which does not need any configuration by operators at all.

Reply:
I've diverted the discussion to GH-133. Thanks for listening.